MINOR: Link to related class in the Admin client documentation

jsancio opened the pull request:
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

jsancio committed:
MINOR: Link to related class in the Admin client documentation

jsancio commented:
Verified that the correct Java Doc was generated.

hachikuji commented on core/src/main/scala/kafka/controller/ControllerContext.scala:
```diff
@@ -37,6 +37,12 @@ object ReplicaAssignment {
+
+/**
+ * @param replicas the seq of brokers assigned to the partition
+ * @param addingReplicas the replicas that were added if there is a pending reassignment
```
nit: "were added" -> "are being added"?

jsancio replied:
Done.

hachikuji commented on core/src/main/scala/kafka/controller/ControllerContext.scala:
```diff
@@ -37,6 +37,12 @@ object ReplicaAssignment {
 
+
+/**
+ * @param replicas the seq of brokers assigned to the partition
```
Maybe worth mentioning that this list includes both the adding and removing replicas?

jsancio replied:
Done.

cmccabe commented:
Looks good overall

jsancio committed:
Better comments for ReplicaAssignment

Include information on how the replicas field relates to addingReplicas
and removingReplicas fields

jsancio commented:
Thanks for the review @cmccabe and @hachikuji. Added a commit based on your feedback.

hachikuji reviewed (approved):
LGTM

hachikuji commented:
I will go ahead and merge this since we are just modifying the comments. I verified locally docs built locally.
